src/components/AnswerForm/index.js

```diff
@@ -1,61 +1,162 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
 import { useParams } from 'react-router-dom';
 
-import { createAnswer } from '@api';
+import { createAnswer, getProductById } from '@api';
 
-const AnswerForm = ({ question_id }) => {
-  const [formData, setFormData] = useState({});
+import Button from '@components/Button';
+import { StyledForm, StyledFormRow, StyledLabel } from './styles';
 
-  const doHandleChange = (e) => {
-    setFormData({
-      ...formData,
-      [e.target.name]: e.target.value,
-    });
-  };
+const AnswerFrom = ({ question, setIsShowing }) => {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm();
+  const { id } = useParams();
+  const [productName, setProductName] = useState('');
+
+  useEffect(async () => {
+    async function fetchData() {
+      const response = await getProductById(id);
+      setProductName(response.data.name);
+    }
+    fetchData();
+  }, [id]);
 
-  const doHandleSubmit = async (e) => {
-    e.preventDefault();
+  const onSubmit = async (data) => {
     try {
-      const response = await createAnswer(formData, question_id);
+      const response = await createAnswer(data, question.question_id);
```

Comment:
same refactor would happen here, where only things being returned are specific to whatever the function is. all parsing and domain-specific logic is encapsulated within the function itself